### PR TITLE
docs: clarify rationales for oauth authentication

### PR DESCRIPTION
**Pull request by gberche-orange**
## Pull request description 

docs: clarify rationales for oauth authentication

Thanks to @vsukhin for his related support in this thread https://discord.com/channels/884464549347074049/885185660808474664/1088040873465487390 

Disclaimer: I'm still a testkube newcomer, I hope did not misunderstood things 

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by aabedraba — APPROVED** [review]

**Review by aabedraba — APPROVED** [review]
Thanks SO much for this. 
LGTM

**Review by vsukhin — APPROVED** [review]

**Review by nicufk — APPROVED** [review]
LGTM, I would prefer @jfermi to take a look at it too

File: docs/docs/getting-started/index.md [hunk 1]
```diff
@@ -9,3 +9,11 @@ In this section you will:
 You can also see the full installation video from our product experts: [Testkube Installation Video](https://www.youtube.com/watch?v=bjQboi3Etys):
 
 <iframe width="100%" height="315" src="https://www.youtube.com/embed/ynzEkOUhxKk" title="YouTube Tutorial: Getting started with Testing in Kubernetes Using Testkube" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
+
+As a summary, you now have a ready testkube setup in your kubernetes cluster, ready to discover the testkube features:
```
Comment (jfermi) [inline docs/docs/getting-started/index.md:13]:
```suggestion
In summary, you now have Testkube setup in your Kubernetes cluster, ready to discover the Testkube features:
```

File: docs/docs/getting-started/index.md [hunk 2]
```diff
@@ -9,3 +9,11 @@ In this section you will:
...
+- the testkube cli enables to locally port forward to the kube dashboard deployment
+- the testkube cli enables to interact with testkube server components through k8s CRs or REST api calls made through the [kube apiserver proxy](https://kubernetes.io/docs/concepts/cluster-administration/proxies/)
```
Comment (jfermi) [inline docs/docs/getting-started/index.md:15]:
```suggestion
- The Testkube CLI allows you to interact with the Testkube server components through k8s CRs or REST API calls made through the [kube apiserver proxy](https://kubernetes.io/docs/concepts/cluster-administration/proxies/).
```

File: docs/docs/getting-started/index.md [hunk 3]
```diff
@@ -9,3 +9,11 @@ In this section you will:
...
+
+As usage of testkube grows within your team, you may choose to:
```
Comment (jfermi) [inline docs/docs/getting-started/index.md:17]:
```suggestion
As usage of Testkube grows within your team, you may choose to:
```

File: docs/docs/getting-started/index.md [hunk 4]
```diff
@@ -9,3 +9,11 @@ In this section you will:
...
+* leverage [managed testkube cloud](../testkube-cloud/intro.md)
```
Comment (jfermi) [inline docs/docs/getting-started/index.md:18]:
```suggestion
* Leverage [managed Testkube cloud](../testkube-cloud/intro.md).
```

File: docs/docs/getting-started/index.md [hunk 5]
```diff
@@ -9,3 +9,11 @@ In this section you will:
...
+* [getting to production](../guides/going-to-production) with your own testkube installation
```
Comment (jfermi) [inline docs/docs/getting-started/index.md:19]:
```suggestion
* [Move to production](../guides/going-to-production) with your own Testkube installation.
```

File: docs/docs/guides/going-to-production/authentication/oauth-cli.md [hunk 6]
```diff
@@ -1,5 +1,9 @@
 # OAuth for CLI
 
+By default, testkube cli uses the "proxy" client which leverages the [kube apiserver proxy](https://kubernetes.io/docs/concepts/cluster-administration/proxies/) to reach the testkube api server. This implies granting users access to testkube internals in order to reach the testkube api REST endpoint (when using commands such as `testkube get artifact` )
```
Comment (jfermi) [inline docs/docs/guides/going-to-production/authentication/oauth-cli.md:3]:
```suggestion
By default, Testkube CLI uses the "proxy" client which leverages the [kube apiserver proxy](https://kubernetes.io/docs/concepts/cluster-administration/proxies/) to reach the Testkube API server. This implies granting users access to Iestkube internals in order to reach the Testkube APIREST endpoint (when using commands such as `testkube get artifact` ).
```

File: docs/docs/guides/going-to-production/authentication/oauth-cli.md [hunk 7]
```diff
@@ -1,5 +1,9 @@
...
+
+This section describes how to protect the testkube api-server REST api endpoint with [oauth2 authentication authorization grant](https://oauth.net/2/grant-types/authorization-code/), for use by the testkube cli as an oauth 2 client. In this mode, kubetest users do not need to be granted "kubetest administrator roles".
```
Comment (jfermi) [inline docs/docs/guides/going-to-production/authentication/oauth-cli.md:5]:
```suggestion
This section describes how to protect the Testkube api-server REST API endpoint with [oauth2 authentication authorization grant](https://oauth.net/2/grant-types/authorization-code/), for use by the Testkube CLI as an oauth 2 client. In this mode, Testkube users do not need to be granted "Testkube administrator roles".
```

File: docs/docs/guides/going-to-production/authentication/oauth-cli.md [hunk 8]
```diff
@@ -21,6 +25,14 @@ Pay attention to the usage of the scheme (http or https) in URIs.
 
 ## Create Github OAuth Application
 
+Currently, only github OAuth authentication is supported. It is not yet possible to configure kube api-server to authenticate testkube cli oauth2 against other Oauth2 IDPs.
```
Comment (jfermi) [inline docs/docs/guides/going-to-production/authentication/oauth-cli.md:28]:
```suggestion
Currently, only GitHub OAuth authentication is supported. It is not yet possible to configure kube api-server to authenticate Testkube CLI OAuth2 against other OAuth2 IDPs. 
```

File: docs/docs/guides/going-to-production/authentication/oauth-cli.md [hunk 9]
```diff
@@ -21,6 +25,14 @@ Pay attention to the usage of the scheme (http or https) in URIs.
 
 ## Create Github OAuth Application
 
+Currently, only github OAuth authentication is supported. It is not yet possible to configure kube api-server to authenticate testkube cli oauth2 against other Oauth2 IDPs. 
+
+In [oauth terminology](https://www.rfc-editor.org/rfc/rfc6749#section-1.1):
```
Comment (jfermi) [inline docs/docs/guides/going-to-production/authentication/oauth-cli.md:30]:
```suggestion
In [OAuth terminology](https://www.rfc-editor.org/rfc/rfc6749#section-1.1): 
```

File: docs/docs/guides/going-to-production/authentication/oauth-cli.md [hunk 10]
```diff
@@ -21,6 +25,14 @@ Pay attention to the usage of the scheme (http or https) in URIs.
 
 ## Create Github OAuth Application
 
+Currently, only github OAuth authentication is supported. It is not yet possible to configure kube api-server to authenticate testkube cli oauth2 against other Oauth2 IDPs. 
+
+In [oauth terminology](https://www.rfc-editor.org/rfc/rfc6749#section-1.1): 
+- github is the *authorization server*.
```
Comment (jfermi) [inline docs/docs/guides/going-to-production/authentication/oauth-cli.md:31]:
```suggestion
- GitHub is the *authorization server*.
```

File: docs/docs/guides/going-to-production/authentication/oauth-cli.md [hunk 11]
```diff
@@ -21,6 +25,14 @@ Pay attention to the usage of the scheme (http or https) in URIs.
...
+- testkube cli is the *client* receiving http redirects from authorization server on a local http endpoint (http://127.0.0.1:13254) served by the cli. A local web browser invoked by [xdg-open](https://linux.die.net/man/1/xdg-open) is required to access github web ui, and then follow http redirect to the local http authorization callback endpoint.
```
Comment (jfermi) [inline docs/docs/guides/going-to-production/authentication/oauth-cli.md:32]:
```suggestion
- Testkube CLI is the *client* receiving HTTP redirects from the authorization server on a local HTTP endpoint (http://127.0.0.1:13254) served by the CLI. A local web browser invoked by [xdg-open](https://linux.die.net/man/1/xdg-open) is required to access GitHub web UI, and then follow HTTP redirect to the local HTTP authorization callback endpoint.
```

File: docs/docs/guides/going-to-production/index.md [hunk 12]
```diff
@@ -0,0 +1,7 @@
+# Overview
+
+This sections provides guidance to get testkube to production, in particular enable access to testkube without requiring k8s privileges to testkube k8s namespace and workloads:
```
Comment (jfermi) [inline docs/docs/guides/going-to-production/index.md:3]:
```suggestion
This sections provides guidance to get Testkube to production, in particular to enable access to Testkube without requiring k8s privileges to the Testkube k8s namespace and workloads:
```

File: docs/docs/guides/going-to-production/index.md [hunk 13]
```diff
@@ -0,0 +1,7 @@
...
+1. Exposing Testkube Dashboard [externally with ingresses](exposing-testkube/overview.md)
```
Comment (jfermi) [inline docs/docs/guides/going-to-production/index.md:4]:
```suggestion
1. Exposing Testkube Dashboard [externally with ingresses](exposing-testkube/overview.md).
```

File: docs/docs/guides/going-to-production/index.md [hunk 14]
```diff
@@ -0,0 +1,7 @@
+# Overview
+
+This sections provides guidance to get testkube to production, in particular enable access to testkube without requiring k8s privileges to testkube k8s namespace and workloads:
+1. Exposing Testkube Dashboard [externally with ingresses](exposing-testkube/overview.md) 
+2. Overall [testkube deployment on AWS](aws.md)
```
Comment (jfermi) [inline docs/docs/guides/going-to-production/index.md:5]:
```suggestion
2. Overall [Testkube deployment on AWS](aws.md).
```

File: docs/docs/guides/going-to-production/index.md [hunk 15]
```diff
@@ -0,0 +1,7 @@
...
+3. Add [Oauth authentication to testkube dashboard](authentication/oauth-ui.md)
```
Comment (jfermi) [inline docs/docs/guides/going-to-production/index.md:6]:
```suggestion
3. Add [OAuth authentication to the Testkube dashboard](authentication/oauth-ui.md). 
```

File: docs/docs/guides/going-to-production/index.md [hunk 16]
```diff
@@ -0,0 +1,7 @@
+# Overview
+
+This sections provides guidance to get testkube to production, in particular enable access to testkube without requiring k8s privileges to testkube k8s namespace and workloads:
+1. Exposing Testkube Dashboard [externally with ingresses](exposing-testkube/overview.md) 
+2. Overall [testkube deployment on AWS](aws.md)
+3. Add [Oauth authentication to testkube dashboard](authentication/oauth-ui.md) 
+4. Add [Oauth authentication to testkube api-server used by cli](authentication/oauth-cli.md) as an alternative to default `proxy` mode leveraging [kube apiserver proxy](https://kubernetes.io/docs/concepts/cluster-administration/proxies/)
```
Comment (jfermi) [inline docs/docs/guides/going-to-production/index.md:7]:
```suggestion
4. Add [OAuth authentication to the Testkube api-server used by the CLI](authentication/oauth-cli.md) as an alternative to default `proxy` mode leveraging [kube apiserver proxy](https://kubernetes.io/docs/concepts/cluster-administration/proxies/).   
```

File: docs/docs/reference/architecture.md [hunk 17]
```diff
@@ -15,6 +15,8 @@ This diagram was made with the C4 diagram technique
 
 ![testkube container diagram](../img/containers.png)
 
+See also [Dependencies for Testkube](../concepts/dependencies.md) mentionning nats, minio and mongodb dependencies
```
Comment (jfermi) [inline docs/docs/reference/architecture.md:18]:
```suggestion
See also [Dependencies for Testkube](../concepts/dependencies.md) describing NATS, MinIO and MongoDB dependencies.
```

**Comment by gberche-orange** [comment]
thanks @jfermi for the improvements and @vsukhin for prompt merging !

**Pull request merged**